br/pkg/utils/misc.go

```diff
@@ -239,6 +241,19 @@ func FlattenValues[K comparable, V any](m map[K][]V) []V {
 	return result
 }
 
+// GetPartitionByName gets the partition ID from the given tableInfo if its name matches
+func GetPartitionByName(tableInfo *model.TableInfo, name ast.CIStr) (int64, error) {
```

Comment:
is there no exiting method to get partition by name?

Reply:
Yes. Replace it by (*Partition).GetPartitionIDByName and reserve the function to convert the returned result `-1` to error.